app/graphql/types/profile.rb

```diff
@@ -66,37 +54,55 @@ def rules(args = {})
     field :compliant_host_count, Int, null: false
 
     def compliant_host_count
-      object.hosts.count { |host| object.compliant?(host) }
+      CollectionLoader.for(object.class, :hosts).load(object).then do |hosts|
+        hosts.count { |host| object.compliant?(host) }
+      end
     end
 
     def total_host_count
-      object.hosts.count
+      CollectionLoader.for(object.class, :hosts).load(object).then(&:count)
     end
 
     def compliant(args = {})
-      host_results = latest_test_result(args)&.rule_results
-      host_results.present? &&
-        (
-          host_results.where(result: 'pass').count /
-          host_results.count.to_f
-        ) >=
-          (object.compliance_threshold / 100.0)
+      latest_test_result_batch(args).then do |latest_test_result|
+        host_results = latest_test_result&.rule_results
+        host_results.present? &&
+          (
+            host_results.where(result: 'pass').count /
+            host_results.count.to_f
+          ) >=
+            (object.compliance_threshold / 100.0)
```

Comment:
I thought we no longer need to do this kind of calculation since we have the TestResult, right?

Reply:
That's right - I had this previous to this change, I'll change it to `latest_test_result.score >= compliance_threshold`